src/functions.php

```diff
@@ -205,17 +205,28 @@ function resolve_uri($id, $parentScope)
     $parts = Uri\parse($id);
 
     if (!empty($parts['path'])) {
-        // If the path ends in a slash, it shouldn't be replaced but instead appended to.
+        // If the original path ends in a slash, it shouldn't be replaced but instead appended to.
```

Comment:
Did you consider `Sabre\Uri\resolve` for this? Curious if it's not good enough, or if there's something I can improve